pkg/reconciler/common/releases.go

```diff
@@ -63,6 +65,52 @@ func InstalledManifest(instance v1alpha1.KComponent) (mf.Manifest, error) {
 	return TargetManifest(instance)
 }
 
+// IsUpDowngradeEligible returns the bool indicate whether the installed manifest is able to upgrade or downgrade to
+// the target manifest.
+func IsUpDowngradeEligible(instance v1alpha1.KComponent) bool {
+	current := instance.GetStatus().GetVersion()
+	// If there is no manifest installed, return true, because the target manifest is able to install.
+	if current == "" {
+		return true
+	}
+	current = sanitizeSemver(current)
+	target := sanitizeSemver(TargetVersion(instance))
+
+	currentMajor := semver.Major(current)
+	targetMajor := semver.Major(target)
+	if currentMajor != targetMajor {
+		// All the official releases of Knative are under the same Major version number. If target and current versions
+		// are different in terms of major version, upgrade or downgrade is not supported.
+		// TODO We need to deal with the the case of bumping major version later.
+		return false
+	}
+
+	var err error
+	currentMinor, err := strconv.Atoi(strings.TrimPrefix(semver.MajorMinor(current), currentMajor+"."))
+	if err != nil {
+		return false
+	}
+
+	targetMinor, err := strconv.Atoi(strings.TrimPrefix(semver.MajorMinor(target), targetMajor+"."))
```

Comment:
Another nit, I'd find the following easier to read:
```suggestion
	targetMinor, err := strconv.Atoi(strings.Split(semver.MajorMinor(target), ".")[1])
```

Or even:

```suggestion
	targetMinor, err := strconv.Atoi(strings.Split(target, ".")[1])
```